### DISCUSSION TITLE
Error on upgrade digitalocean App Platform deployment [Sep 28]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#3946] [Discord Threads]
> 🙋‍♂️ **charlietrigger**

**Comment by o1lab** [comment]
> 🙋‍♂️ **charlietrigger**  
#### This is the error I am getting  
![image0](https://cdn.discordapp.com/attachments/1022113707679027200/1022114867446030376/unknown.png)  
#  
> 🙋‍♂️ **charlietrigger**  
#### To add to this: I have tried buildinga new app with a fresh database (a managed db mysql on digitalocean) and using the environment variable NC_DB set from the first deployment. This results in a similar migration failed error where if cannot create table 'nc_projects'. Am I missing something here?   
####    
#### This is the most telling error I think:   
#### 2-09-21 12:45:36]   sqlMessage: "Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set. Add a primary key to the table or unset this variable to avoid this message. Note that tables without a primary key can cause performance problems in row-based replication, so please consult your DBA before changing this setting.",   
#### [nocodb-nocodb] [2022-09-21 12:45  
#  
> 🙋‍♂️ **charlietrigger**  
#### After reading up for a bit I found that on a  digitalocean managed mysql database sql_require_primary_key is set to true on default. I requires api access to change this setting. They warn that setting this to false can lead to issues:   
####    
#### 'For clusters with more than one node, we strongly recommend keeping sql_require_primary_key as True. Setting it to False may severely impact replication and cause issues.'   
####    
#### Would it be possible to add a primary key to the nc_projects table?  
#  
> 🙋‍♂️ **charlietrigger**  
#### I have found a fix as described here:   
#### https://stackoverflow.com/questions/62418099/unable-to-create-or-change-a-table-without-a-primary-key-laravel-digitalocean/72438019#72438019   
####    
#### You can change your db setting to sql_require_primary_key to false. Do the update and if you want to you can set it back to true. This works like a charm and now the app platform deploy runs without issues and without resetting project.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1022113707679027200)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
